src/Tes.Runner/Storage/TerraSasResolutionStrategy.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+using Azure.Storage.Blobs;
+using Azure.Storage.Sas;
+using Microsoft.Extensions.Logging;
+using Tes.ApiClients;
+using Tes.ApiClients.Models.Terra;
+using Tes.Runner.Models;
+using Tes.Runner.Transfer;
+using TesApi.Web.Management.Models.Terra;
+
+namespace Tes.Runner.Storage
+{
+    public class TerraSasResolutionStrategy : ISasResolutionStrategy
+    {
+        private const int TokenExpirationInSeconds = 3600 * 48; //48 hours, max Azure Batch node runtime. 
+        private readonly TerraWsmApiClient terraWsmApiClient;
+        private readonly TerraRuntimeOptions terraRuntimeOptions;
+        private const string LzStorageAccountNamePattern = "lz[0-9a-f]*";
```

Comment:
can we add a comment that points to the line of code in the LZ repo that defines this, if possible?